subsys/bluetooth/controller/ll_sw/ull_sync.c

```diff
@@ -203,22 +202,29 @@ uint8_t ll_sync_create_cancel(void **rx)
 		return BT_HCI_ERR_CMD_DISALLOWED;
 	}
 
-	node_rx = (void *)scan->per_scan.node_rx_estab;
-	link_sync_estab = node_rx->hdr.link;
-	link_sync_lost = sync->node_rx_lost.link;
+	se = (void *)scan->per_scan.node_rx_estab;
+	link_sync_estab = se->hdr.link;
+	link_sync_lost = sync->node_rx_lost.hdr.link;
 
 	ll_rx_link_release(link_sync_lost);
 	ll_rx_link_release(link_sync_estab);
-	ll_rx_release(node_rx);
+	ll_rx_release(se);
+
+	se = (void *)&sync->node_rx_lost;
+	se->hdr.type = NODE_RX_TYPE_SYNC;
+	se->hdr.handle = 0xffff;
 
-	node_rx = (void *)&sync->node_rx_lost;
-	node_rx->hdr.type = NODE_RX_TYPE_SYNC;
-	node_rx->hdr.handle = 0xffff;
-	node_rx->hdr.rx_ftr.param = sync;
-	se = (void *)node_rx->pdu;
-	se->status = BT_HCI_ERR_OP_CANCELLED_BY_HOST;
+	/* NOTE: struct node_rx_lost has uint8_t member following the
+	 *       struct node_rx_hdr to store the reason.
+	 */
+	*((uint8_t *)se->pdu) = BT_HCI_ERR_OP_CANCELLED_BY_HOST;
```

Comment:
This is quite ugly. I would prefer to have the type of the `node_rx_lost` field be non-anonymous so `se` can be declared with that type and the `reason` field is accessible. I would also be okay with accessing `sync->node_rx_lost.reason` directly.